### ISSUE TITLE
Document @UIType consequences, and other GUI Effect Checker policies more clearly

### ISSUE THREAD

**Issue by csgordon** [typetools/checker-framework#4698]
There are reports that the documentation for ```@UIType``` is unclear with regards to how it interacts with inheritance (inherited by anonymous inner classes, not inherited by named subtypes).  Need to audit the docs for the checker and make sure this, and other policies, are both clear and easy to find.

**Comment by csgordon** [comment]
More generally, this should include additional documentation on the core notion of effect systems, which currently have only a bare-minimum explanation in the manual
